### PR TITLE
feat(Core/Creature): Implement c_t_r & c_t_s

### PR DESCRIPTION
**Pull request by Kitzunu**
* https://github.com/TrinityCore/TrinityCore/commit/d5fb0a30ec62f3799416326d81d5d14992f9a6df

Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>

## Changes Proposed:
-  Move creature resistance to creature_template_resistance 
-  Move creature spells to creature_template_spell 
-  rename CREATURE_MAX_SPELLS to MAX_CREATURE_SPELLS

## Tests Performed:
- Builds
- Tested in-game as shown below

* creature_template_resistance
image
image
image
image
image
image

* creature_template_spell
image

## How to Test the Changes:

* creature_template_resistance
```sql
DELETE FROM creature_template_resistance WHERE CreatureID = 299;
INSERT INTO creature_template_resistance (CreatureID, School, Resistance) VALUES
(299, 1, 5000),
(299, 2, 5000),
(299, 3, 5000),
(299, 4, 5000),
(299, 5, 5000),
(299, 6, 5000);
```
1. .go c id 299
1. Test casting spells with the different schools. (HOLY, FIRE, NATURE, FROST, SHADOW, ARCANE)

* creature_template_spell

1. Be ally
2. .npc add temp 28468
3. mount the vehicle and see it has spells

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
that crazy PR title

**Comment by Helias** [comment]
This will help to align AC with TC.
This can break something, it's a big change I think, need several test.

**Comment by Kitzunu** [comment]
As far as I have tested it works as intended

**Comment by 55Honey** [comment]
The resistance for direct damage spells seems to work not 100%.
image
image
image
image


I could affect the mob with Entangling roots by buffing myself with Nature's Grasp.
image

Also when i apply a dot like Moonfire, the damage is resisted, but the debuff is still applied. Dot ticks are resisted again most of the time, same as direct-damage spells.
image

Applying thorns to my test toon made the mob receive nature damage.
image

"Stampy" works fine. Spells are there work as in the description.
image
Noteworthy: "Hemet Nesingwary" is fighting with me while im sitting on Stampy.
image

Funny: When i `.tele` while mounted on "Stampy" the player is dismounted and Stampy stays in place.

File: data/sql/updates/pending_db_world/rev_1611776034016932100.sql [hunk 1]
```diff
@@ -5,16 +5,16 @@ CREATE TABLE IF NOT EXISTS `creature_template_resistance` (
   `CreatureID` MEDIUMINT(8) UNSIGNED NOT NULL,
   `School` TINYINT(6) UNSIGNED NOT NULL,
   `Resistance` SMALLINT(6) DEFAULT NULL,
-  `VerifiedBuild` SMALLINT(5) DEFAULT 0,
+  `VerifiedBuild` SMALLINT(5) DEFAULT '0',
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611776034016932100.sql:8]: revert this please

File: data/sql/updates/pending_db_world/rev_1611776034016932100.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  PRIMARY KEY (`CreatureID`, `School`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+DROP TABLE IF EXISTS `creature_template_spell`;
+CREATE TABLE IF NOT EXISTS `creature_template_spell` (
+  `CreatureID` MEDIUMINT(8) unsigned NOT NULL,
+  `Index` TINYINT(6) UNSIGNED NOT NULL DEFAULT '0',
+  `Spell` MEDIUMINT(8) UNSIGNED DEFAULT NULL,
+  `VerifiedBuild` SMALLINT(5) DEFAULT '0',
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611776034016932100.sql:17]: also here, why using string if it's an int ?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1611776034016932100.sql:17]: I just copy-pasted from my Heidi. Gonna do it asap

**Comment by FrancescoBorzi** [comment]
this also requires adapting Keira3, otherwise it will be completely broken when you try to edit creatures.

@Kitzunu I can assist you but I can't do all the work on my own because I'm very busy lately

**Comment by Kitzunu** [comment]
I will resolve the conflicts when I asap

**Comment by Kitzunu** [comment]
done

File: src/server/game/Entities/Unit/Unit.h [hunk 3]
```diff
@@ -246,7 +246,7 @@ enum UnitRename
     UNIT_CAN_BE_ABANDONED   = 0x02,
 };
 
-#define CREATURE_MAX_SPELLS     8
+static const uint32 MAX_CREATURE_SPELLS = 8;
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Unit/Unit.h:249]: constexpr

**Comment by Kitzunu** [comment]
LF review q.q

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**